tests/applications/test_junebug_message_api.py

```diff
@@ -0,0 +1,667 @@
+#         the transport queue if the config option is set.'''
+#         worker = yield self.get_worker(config={
+#             'message_queue': 'testqueue'
+#         })
+#         msg = TransportUserMessage.send(to_addr='+1234', content='testcontent')
+#         yield self.app_helper.dispatch_outbound(
+#             msg, connector_name='testqueue')
+
+#         [dispatched_msg] = yield self.app_helper.wait_for_dispatched_outbound(
+#             connector_name=worker.transport_name)
+#         self.assertEqual(dispatched_msg, msg)
+
+#     @inlineCallbacks
+#     def test_forward_ack_amqp(self):
+#         '''A sent ack event should be forwarded to the correct AMQP queue if
+#         the config option is set.'''
+#         worker = yield self.get_worker(config={
+#             'message_queue': 'testqueue'
+#         })
+#         event = TransportEvent(
+#             event_type='ack',
+#             user_message_id='msg-21',
+#             sent_message_id='msg-21',
+#             timestamp='2015-09-22 15:39:44.827794')
+
+#         yield worker.consume_ack(event)
+
+#         [dispatched_msg] = self.app_helper.get_dispatched(
+#             'testqueue', 'event', TransportEvent)
+
+#         self.assertEqual(dispatched_msg['event_id'], event['event_id'])
+
+#     @inlineCallbacks
+#     def test_forward_nack_amqp(self):
+#         '''A sent nack event should be forwarded to the correct AMQP queue if
+#         the config option is set.'''
+#         worker = yield self.get_worker(config={
+#             'message_queue': 'testqueue'
+#         })
+#         event = TransportEvent(
+#             event_type='nack',
+#             user_message_id='msg-21',
+#             nack_reason='too many foos',
+#             timestamp='2015-09-22 15:39:44.827794')
+
+#         yield worker.consume_nack(event)
+
+#         [dispatched_msg] = self.app_helper.get_dispatched(
+#             'testqueue', 'event', TransportEvent)
+
+#         self.assertEqual(dispatched_msg['event_id'], event['event_id'])
+
+#     @inlineCallbacks
+#     def test_forward_dr_amqp(self):
+#         '''A sent delivery report event should be forwarded to the correct
+#         AMQP queue if the config option is set.'''
+#         worker = yield self.get_worker(config={
+#             'message_queue': 'testqueue'
+#         })
+#         event = TransportEvent(
+#             event_type='delivery_report',
+#             user_message_id='msg-21',
+#             delivery_status='pending',
+#             timestamp='2015-09-22 15:39:44.827794')
+
+#         yield worker.consume_nack(event)
+
+#         [dispatched_msg] = self.app_helper.get_dispatched(
+#             'testqueue', 'event', TransportEvent)
+
+#         self.assertEqual(dispatched_msg['event_id'], event['event_id'])
+
+## Storing inbound messages. As far as I can tell, we only use the message
+## store for event routing. We don't need to store inbound events for that.
```

Comment:
Is it maybe also used for outbound message replies? Sometimes transports rely on the transport metadata and other fields to be copied between the MO and MT messages, also used for routing.

Reply:
I haven't really looked at the outbound code yet, so I'll either implement this if we use it there or remove it if we don't.